File: src/org/lineageos/samsungparts/SamsungPartsService.java
/*
 * Copyright (c) 2021-2025 The LineageOS Project
 * Licensed under the Apache License, Version 2.0
 */

 package org.lineageos.samsungparts;

 import android.app.Service;
 import android.content.Context;
 import android.content.Intent;
 import android.os.IBinder;
 import android.util.Log;
 
 public class SamsungPartsService extends Service {
     private static final String TAG = "SamsungPartsService";
 
     @Override
     public void onCreate() {
         super.onCreate();
         Log.i(TAG, "SamsungPartsService started");
 
         // Start EyeComfortSolutionService from here
         startService(new Intent(this, EyeComfortSolutionService.class));
     }
 
     @Override
     public IBinder onBind(Intent intent) {
         return null;
     }
 }
 

File: src/org/lineageos/samsungparts/SamsungPartsService.kt
/*
 * Copyright (c) 2025 The LineageOS Project
 * Licensed under the Apache License, Version 2.0
 */

package org.lineageos.samsungparts

import android.app.Service
import android.content.Intent
import android.os.IBinder
import android.util.Log
import org.lineageos.samsungparts.tsp.SecTspService
import org.lineageos.samsungparts.udfps.SemUdfpsService

class SamsungPartsService : Service() {
    private val TAG = "SamsungPartsService"

    override fun onCreate() {
        super.onCreate()
        Log.i(TAG, "SamsungPartsService started")

        // Start EyeComfortSolutionService
        Log.i(TAG, "Starting EyeComfortSolutionService...")
        startService(Intent(this, EyeComfortSolutionService::class.java))

        // Start SecTspService
        Log.i(TAG, "Starting SecTspService...")
        startService(Intent(this, SecTspService::class.java))

        // Start SemUdfpsService
        Log.i(TAG, "Starting SemUdfpsService...")
        startService(Intent(this, SemUdfpsService::class.java))
    }

    override fun onBind(intent: Intent?): IBinder? = null
}
